File: src/main/java/quebec/virtualite/backend/services/domain/DomainServiceImpl.java
package quebec.virtualite.backend.services.domain;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import quebec.virtualite.backend.services.domain.database.GreetingRepository;
import quebec.virtualite.backend.services.domain.entities.GreetingEntity;

import java.util.ArrayList;
import java.util.List;

@Service
public class DomainServiceImpl implements DomainService
{
    @Autowired
    private GreetingRepository greetingRepository;

    @Override
    public void deleteGreetings()
    {
        greetingRepository.deleteAll();
    }

    @Override
    public List<GreetingEntity> getGreetings()
    {
        return list(greetingRepository.findAll());
    }

    @Override
    public void recordGreeting(String name)
    {
        GreetingEntity greeting = greetingRepository.findByName(name)
            .orElse(new GreetingEntity()
                .setName(name));

        greetingRepository.save(greeting);
    }

    private <T> List<T> list(Iterable<T> iterable)
    {
        List<T> list = new ArrayList<>();
        iterable.forEach(list::add);

        return list;
    }
}

File: src/main/java/quebec/virtualite/backend/services/domain/DomainServiceImpl.kt
package quebec.virtualite.backend.services.domain

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service
import quebec.virtualite.backend.services.domain.database.GreetingRepository
import quebec.virtualite.backend.services.domain.entities.GreetingEntity

@Service
class DomainServiceImpl : DomainService
{
    @Autowired
    private lateinit var greetingRepository: GreetingRepository

    override fun deleteGreetings()
    {
        greetingRepository.deleteAll()
    }

    override fun getGreetings(): List<GreetingEntity>
    {
        return list(greetingRepository.findAll())
    }

    override fun recordGreeting(name: String)
    {
        val greeting = greetingRepository.findByName(name) ?: GreetingEntity(name)

        greetingRepository.save(greeting)
    }

    private fun <T> list(iterable: Iterable<T>): List<T>
    {
        val list: ArrayList<T> = ArrayList()
        iterable.forEach(list::add)

        return list
    }
}